### PR TITLE
feat(curriculum): update handleclick method to use state function

### LINKED ISSUE #42208: Challenge implies using anti-pattern which may have unexpected results in real situations
**Affected page**
https://www.freecodecamp.org/learn/front-end-libraries/react/render-conditionally-from-props

The lines 242-246 of the [file](https://github.com/freeCodeCamp/freeCodeCamp/blob/4d024f3e4bcbb0a0016e6a6bce01619c25e54dbc/curriculum/challenges/english/03-front-end-libraries/react/render-conditionally-from-props.md#L242-L246) read:
```js
handleClick() {
  this.setState({
    counter: 0 // Change this line
  });
}
```
This implies that you can increment `counter` as follows: `counter: this.state.counter + 1`.
As already taught at this point of the curriculum, this method can result in unexpected behavior, since React may batch state updates for performance reasons.
AFAIK, the following is the correct way to handle this situation:
```js
handleClick() {
  this.setState(prevState => ({counter: prevState.counter + 1}))
}
```
**Expected behavior**
Right now, both methods of incrementing `counter` are accepted. I suggest we change the provided code and the tests so that only the second method is implied and accepted.

**System:**
 - Device: Laptop
 - OS: Ubuntu 20.4
 - Browser: Opera
 - Version: 76

### PR DESCRIPTION
**Pull request by suburban-daredevil**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] The handleClick method is now updated to get the user to use the updater function inside the setState method instead of directly mutating the state of the component

Closes #42208

Updated both the seed code and the answer code to enforce the usage of updater function inside the setState method. This is done in order to maintain the consistency of data inside the state of the component.

### CODE REVIEW

**Comment by lasjorg** [comment]
For the incomplete updater function do we want the empty function body, or not? I'm just thinking about the use of an implicit return and having a starting syntax that might be more "open" to it.

So this:
```
this.setState(prevState => prevState);
```

Instead of this:

```
this.setState(prevState => {

});
```

I'm fine with it either way really, so LGTM.

**Comment by suburban-daredevil** [comment]
Hello @lasjorg. Thanks for going with my pull request. 

Personally as a learner myself, I think having two `prevState` keywords might confuse the learner who is a beginner while learning about the updater function in `this.setState` in React.

So it is my humble suggestion to go with this :

```
this.setState(prevState => {

});
```
However if you want to me to change the code, I am more than happy to do the same. Thanks once again for selecting my PR.

File: curriculum/challenges/english/03-front-end-libraries/react/render-conditionally-from-props.md [hunk 1]
```diff
@@ -240,8 +240,9 @@ class GameOfChance extends React.Component {
     this.handleClick = this.handleClick.bind(this);
   }
   handleClick() {
-    this.setState({
-      counter: 0 // Change this line
+    // Complete the setState updater function
+    this.setState(prevState => {
+
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/03-front-end-libraries/react/render-conditionally-from-props.md:245]:
```suggestion
    this.setState(prevState => {
      // Complete the return statement:
      return {
        counter: prevState
      }
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Hey there,

We are only using a set of [agreed-upon comments](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/dictionaries/english/comments.json), and should not be adding to them (for i18n reasons).

**Comment by suburban-daredevil** [comment]
Hello @ShaunSHamilton,

I have committed the changes that were requested. Thank you!

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 . I am happy with this. Thank you, for making that change.

@lasjorg Feel free to agree/disagree

**Review by lasjorg — APPROVED** [review]
@ShaunSHamilton Didn't know about that. I think the comment was just something I added to the initial code suggestion.

LGTM

**Pull request merged**